### ISSUE TITLE
Unable to connect to the service through ntex-tls when openssl version is 1.0.2u

### ISSUE THREAD

**Issue by jm-observer** [ntex-rs/ntex#152]
hi, when i connect to the google just only through openssl version 1.0.2u, it success, but when i try connnet service through ntex-tls,program error:
```
[2022-12-07 14:57:56.725][main] TRACE [ntex_connect::resolve:37] DNS resolver: resolving host "google.com"
[2022-12-07 14:57:56.730][main] TRACE [ntex_connect::resolve:57] DNS resolver: host "google.com" resolved to [172.217.24.78:443, [2404:6800:4005:80
f::200e]:443]
[2022-12-07 14:57:56.730][main] TRACE [ntex_connect::service:163] TCP connector - connecting to "google.com" port:443
[2022-12-07 14:57:56.730][main] TRACE [mio::poll:532] registering event source with poller: token=Token(0), interests=READABLE | WRITABLE
[2022-12-07 14:57:56.742][main] TRACE [ntex_connect::service:210] TCP connector - successfully connected to connecting to "google.com" - Some(172.2
17.24.78:443)
[2022-12-07 14:57:56.742][main] TRACE [ntex_connect::openssl:56] SSL Handshake start for: "google.com"
[2022-12-07 14:57:56.742][main] TRACE [ntex_tokio::io:347] flushed 517 bytes
[2022-12-07 14:57:56.830][main] TRACE [ntex_io::tasks:57] new 1 bytes available, wakeup dispatcher
[2022-12-07 14:57:56.830][main] TRACE [ntex_io::io:531] waking up io read task
[2022-12-07 14:57:56.833][main] TRACE [ntex_io::tasks:57] new 1 bytes available, wakeup dispatcher
[2022-12-07 14:57:56.981][main] TRACE [ntex_connect::openssl:70] SSL Handshake error: Error { code: ErrorCode(1), cause: Some(Ssl(ErrorStack([Error
 { code: 336085247, library: "SSL routines", function: "ssl3_connect", reason: "unknown state", file: "s3_clnt.c", line: 646 }]))) }
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Custom { kind: Other, error: "error:140840FF:SSL routines:ssl3_connect:u
nknown state:s3_clnt.c:646:" })', src/bin/ntex-connector.rs:21:44
```

[here ](https://github.com/jm-observer/openssl1-0-2u)is my code and the compile env(docker image).

**Comment by fafhrd91** [comment]
you probably miss root certs

**Comment by jm-observer** [comment]
please see my [code](https://github.com/jm-observer/openssl1-0-2u/blob/main/src/bin/ntex-connector.rs#L17), i had set SslVerifyMode = SslVerifyMode:NONE, So it doesn't need a ca file. In fact, I also tested async-native-tls and tokio-native-tls and they both connect correctly 
```
builder.set_verify(ssl::SslVerifyMode::NONE);
```

**Comment by fafhrd91** [comment]
works for me

```
[2022-12-08 09:36:35.688][main] TRACE [ntex_connect::resolve:37] DNS resolver: resolving host "google.com"
[2022-12-08 09:36:35.693][main] TRACE [ntex_connect::resolve:57] DNS resolver: host "google.com" resolved to [64.233.161.102:443, 64.233.161.138:443, 64.233.161.100:443, 64.233.161.139:443, 64.233.161.113:443, 64.233.161.101:443]
[2022-12-08 09:36:35.694][main] TRACE [ntex_connect::service:163] TCP connector - connecting to "google.com" port:443
[2022-12-08 09:36:35.694][main] TRACE [mio::poll:532] registering event source with poller: token=Token(0), interests=READABLE | WRITABLE
[2022-12-08 09:36:35.829][main] TRACE [ntex_connect::service:210] TCP connector - successfully connected to connecting to "google.com" - Some(64.233.161.102:443)
[2022-12-08 09:36:35.829][main] TRACE [ntex_connect::openssl:56] SSL Handshake start for: "google.com"
[2022-12-08 09:36:35.830][main] TRACE [ntex_tokio::io:347] flushed 517 bytes
[2022-12-08 09:36:35.964][main] TRACE [ntex_io::tasks:57] new 1 bytes available, wakeup dispatcher
[2022-12-08 09:36:35.964][main] TRACE [ntex_tokio::io:347] flushed 80 bytes
[2022-12-08 09:36:35.964][main] TRACE [ntex_connect::openssl:66] SSL Handshake success: "google.com"
Connected to ssl server
[2022-12-08 09:36:35.965][main] TRACE [ntex_tokio::io:347] flushed 104 bytes
Send result: ()
[2022-12-08 09:36:35.965][main] TRACE [ntex_io::io:531] waking up io read task
[2022-12-08 09:36:35.965][main] TRACE [ntex_io::io:571] not enough data to decode next frame
[2022-12-08 09:36:36.217][main] TRACE [ntex_io::io:571] not enough data to decode next frame
[2022-12-08 09:36:36.217][main] TRACE [ntex_tokio::io:83] tokio stream is disconnected
[2022-12-08 09:36:36.218][main] TRACE [ntex_io::tasks:57] new 927 bytes available, wakeup dispatcher
[2022-12-08 09:36:36.219][main] TRACE [ntex_tokio::io:204] write task is instructed to terminate
[2022-12-08 09:36:36.223][main] TRACE [mio::poll:663] deregistering event source from poller
Received: b"HTTP/1.1 301 Moved Permanently\r\nLocation: https://www.google.com/\r\nContent-Type: text/html; charset=UTF-8\r\nCross-Origin-Opener-Policy-Report-Only: same-origin-allow-popups; report-to=\"gws\"\r\nReport-To: {\"group\":\"gws\",\"max_age\":2592000,\"endpoints\":[{\"url\":\"https://csp.withgoogle.com/csp/report-to/gws/other\"}]}\r\nDate: Thu, 08 Dec 2022 08:36:36 GMT\r\nExpires: Sat, 07 Jan 2023 08:36:36 GMT\r\nCache-Control: public, max-age=2592000\r\nServer: gws\r\nContent-Length: 220\r\nX-XSS-Protection: 0\r\nX-Frame-Options: SAMEORIGIN\r\nAlt-Svc: h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\"\r\nConnection: close\r\n\r\n<HTML><HEAD><meta http-equiv=\"content-type\" content=\"text/html;charset=utf-8\">\n<TITLE>301 Moved</TITLE></HEAD><BODY>\n<H1>301 Moved</H1>\nThe document has moved\n<A HREF=\"https://www.google.com/\">here</A>.\r\n</BODY></HTML>\r\n"
disconnecting
[2022-12-08 09:36:36.224][main] TRACE [ntex_io::io:706] io is dropped, force stopping io streams IO_STOPPED | IO_STOPPING | IO_STOPPING_FILTERS | RD_READY
[2022-12-08 09:36:36.224][main] TRACE [ntex_io::ioref:62] force close io stream object
```
